### PR TITLE
Add signals to filter participant list display

### PR DESCRIPTION
**Pull request by SegiNyn**
Hello, we would like to add 3 signals for filtering registrations and regforms in the participant list

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Please send new PRs against master :)

**Comment by OmeGak** [comment]
@SegiNyn, could you rewrite this PR once #4981 is merged? Rather than adding new signals, we can simply add the `.signal_query(...)` call to the queries we want to intercept.

```python
query = (Registration.query.with_parent(self.event)
         .filter(Registration.is_state_publishable, ~RegistrationForm.is_deleted)
         .join(Registration.registration_form)
         .options(subqueryload('data').joinedload('field_data'),
                  contains_eager('registration_form'))
         .signal_query('publishable-registrations-in-participant-list'))
```

On the plugin side, we can intercept them like this:

```python
@signals.core.db_query.connect_via('publishable-registrations-in-participant-list')
def _intercept_publishable_registrations_in_participant_list(sender, **kwarg):
    query = kwarg['query']
    return query.filter(...)
```

**Comment by ThiefMaster** [comment]
I think she can already start rewriting it on top of your PR ;)

File: indico/modules/events/registration/controllers/display.py [hunk 1]
```diff
@@ -142,7 +142,8 @@ def _deduplicate_reg_data(reg_data_iter):
                  .filter(Registration.is_state_publishable, ~RegistrationForm.is_deleted)
                  .join(Registration.registration_form)
                  .options(subqueryload('data').joinedload('field_data'),
-                          contains_eager('registration_form')))
+                          contains_eager('registration_form'))
+                 .signal_query('publishable-registrations-in-merged-participant-list'))
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/display.py:146]:
```suggestion
                 .signal_query('merged-participant-list-publishable-registrations'))
```

File: indico/modules/events/registration/controllers/display.py [hunk 2]
```diff
@@ -188,7 +189,8 @@ def _sort_key_date(column_id):
                  .options(subqueryload('data'))
                  .order_by(db.func.lower(Registration.first_name),
                            db.func.lower(Registration.last_name),
-                           Registration.friendly_id))
+                           Registration.friendly_id)
+                 .signal_query('publishable-registrations-in-participant-list', regform=regform))
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/display.py:193]:
```suggestion
                 .signal_query('participant-list-publishable-registrations', regform=regform))
```

File: indico/modules/events/registration/controllers/display.py [hunk 3]
```diff
@@ -202,6 +204,7 @@ def _process(self):
         regforms = (RegistrationForm.query.with_parent(self.event)
                     .filter(RegistrationForm.is_participant_list_visible(self.event.is_user_registered(session.user)))
                     .options(subqueryload('registrations').subqueryload('data').joinedload('field_data'))
+                    .signal_query('publishable-regforms-in-participant-list')
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/display.py:207]:
```suggestion
                    .signal_query('participant-list-publishable-regforms')
```

**Review by ThiefMaster — COMMENTED** [review]
Those names are a bit more concise

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**